demo/app/controllers/demo_controller.rb

```diff
@@ -0,0 +1,151 @@
+class DemoController < ApplicationController
+
+  before_action :validate_config!, only: :create
+  before_action :get_visitor, only: [:shop,:buy]
+  before_action :get_project_configuration, only: [:shop,:buy]
+  before_action :check_optimizely_client, only: [:shop,:buy]
+  before_action :get_product, only: [:buy]
+
+  def new
+    # Finds Project ID if stored in session.
+    # Returns new config object if Project ID not found
+    #   else Finds config by Project ID stored in session
+    #   Initializes OptimizelyService and generates Optimizely client object
+    # Returns config
+    
+    if session[:config_project_id].present?
+      @config = Config.find_by_project_id(session[:config_project_id])
+      get_or_generate_optimizely_client
+    else
```

Comment:
You can remove this else block because 
get_or_generate_optimizely_client will return Config.new if @config is null

Reply:
As discussed. if we remove this we need to set Config new in case session is empty.